lib/internal/Magento/Framework/App/FeedFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\App;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Feed factory
+ */
+class FeedFactory implements FeedFactoryInterface
+{
+    /**
+     * @var \Zend_Feed
+     */
+    private $feedProcessor;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param \Zend_Feed $feedProcessor
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        \Zend_Feed $feedProcessor,
+        LoggerInterface $logger
+    ) {
+        $this->feedProcessor = $feedProcessor;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Get a new \Magento\Framework\App\Feed object from a custom array
+     *
+     * @throws \Magento\Framework\Exception\InputException
+     * @param array $data
+     * @param string $format
+     * @return \Magento\Framework\App\FeedInterface
+     */
+    public function create(
+        array $data,
+        $format = FeedFactoryInterface::DEFAULT_FORMAT
+    ) {
+        try {
+            return $this->feedProcessor->importArray($data, $format);
```

Comment:
Return type of feedProcessor is not compatible with the return type of this method (`FeedInterface` !== `Zend_Feed_Abstract`)